src/class-wpml-beaver-builder-register-strings.php

```diff
@@ -20,4 +21,47 @@ protected function register_strings_for_modules( array $data_array, array $packa
 			}
 		}
 	}
+
+	/**
+	 * The modules are not in the order they appear on the page,
+	 * so we need to sort it before to register the strings.
+	 *
+	 * @param array $modules
+	 *
+	 * @return array
+	 */
+	private function sort_modules_before_string_registration( array $modules ) {
+		uasort( $modules, array( $this, 'sort_modules_by_position_only' ) );
+		return $this->sort_modules_by_parent_and_child( $modules );
+	}
+
+	/**
+	 * @param array        $all_modules
+	 * @param string|null  $parent_hash
+	 * @param array        $sorted_modules
+	 *
+	 * @return array
+	 */
+	private function sort_modules_by_parent_and_child( array $all_modules, $parent_hash = null, array $sorted_modules = array() ){
```

Comment:
@strategio `$all_modules` makes me think the input has something special compared to `$modules` used in `sort_modules_before_string_registration`, but instead, it's the same data: can't you use the same name everywhere?

Reply:
Oh, now I think I get it.

These are the "flat" representation of the modules and what `sort_modules_by_parent_and_child` seem to do is creating some hierarchy sorted version of the array. Am I right?

I think a description of this method, in the PHPDoc, is badly needed.

Reply:
Yes @andreasciamanna, we get a flat tree of modules. The unit tests demonstrates this https://github.com/OnTheGoSystems/wpml-page-builders-beaver-builder/pull/16/files#diff-77dbb17f691cc1dc504c62173c5c2345R128. I added a comment to make this clearer.